src/System/Console/Hawk/Lock.hs

```diff
@@ -20,106 +20,32 @@ module System.Console.Hawk.Lock
     , withTestLock
     ) where
 
-import Control.Concurrent ( threadDelay )
-import Control.Exception
-import Control.Monad ( when, guard )
-import Data.List ( elemIndex )
-import GHC.IO.Exception
-import GHC.IO.Handle ( Handle, hGetContents, hClose )
-import Network.BSD ( getProtocolNumber ) -- still cross-platform, don't let the name fool you
-import Network ( PortID (..), connectTo )
-import Network.Socket
-import Text.Printf
-
-
--- use a socket number as a lock indicator.
-withSocketLock :: Bool -> IO a -> IO a
-withSocketLock testing = withSocketsDo . bracket (lock testing) unlock . const
-
--- make sure GHC inlines withSocketLock in the following two functions
--- and optimizes the testing and non-testing versions separately.
-{-# INLINE withSocketLock #-}
+import System.Directory ( doesFileExist, removeFile )
+import Control.Monad ( when )
+import System.FileLock ( SharedExclusive( Exclusive ), withFileLock )
 
+-- | Create a temporary file to indicate to other hawk process
+--  not to run until this process is unblocked
 withLock :: IO a -> IO a
-withLock = withSocketLock False
+withLock action = withFileLock lockfile Exclusive (const action) >>= cleanup
```

Comment:
The name "cleanup" makes it sound like this is something which should be executed even if the body is interrupted by an exception, for example if the user presses `CTRL-C`. Would it make sense to use `finally` to ensure it is always executed?

Reply:
Actually, now that I look at the implementation of `cleanup`, I see that this deletes the file, which forcibly releases the lock. We wouldn't want killing one hawk process which is still waiting to acquire the lock to cause the lock held by another hawk process to be forcibly released.

In fact, if we put the lock file in hint's context directory as I suggest, we can simply leave the file there, no need to delete it afterwards.